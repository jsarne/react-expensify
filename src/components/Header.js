import React from 'react';
import {connect} from 'react-redux';
import {NavLink} from 'react-router-dom';
import {startLogout} from '../actions/auth';

export const Header = ({logout}) => (
  <header>
    <h1>Expensify</h1>
    <NavLink to="/" activeClassName="is-active" exact={true}>Home</NavLink>
    <NavLink to="/create" activeClassName="is-active">Add Expense</NavLink>
    <button onClick={logout}>Logout</button>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  logout: () => dispatch(startLogout())
});
export default connect(undefined, mapDispatchToProps)(Header);