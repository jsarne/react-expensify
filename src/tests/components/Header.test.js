// react-test-renderer version
// import React from 'react';
// import ReactShallowRenderer from 'react-test-renderer/shallow';
// import Header from '../../components/Header';

// test('header render', () => {
//   const renderer = new ReactShallowRenderer();
//   renderer.render(<Header />);
//   expect(renderer.getRenderOutput()).toMatchSnapshot();
// });


// enzyme version
import React from 'react';
import {shallow} from 'enzyme';
//import toJSON from 'enzyme-to-json';  // can skip this b/c we loaded this initializer in Jest config
import Header from '../../components/Header';

test('header render', () => {
  const wrapper = shallow(<Header />);
  //expect(wrapper.find('h1').text()).toBe("Expensify");
  //expect(toJSON(wrapper)).toMatchSnapshot();
  expect(wrapper).toMatchSnapshot();
});
