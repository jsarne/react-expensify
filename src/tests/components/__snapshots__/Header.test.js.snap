// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header render 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    to="/dashboard"
  >
    Home
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/create"
  >
    Add Expense
  </NavLink>
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</header>
`;
